src/cascadia/TerminalSettingsEditor/Profiles.cpp

```diff
@@ -637,19 +628,42 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         _State.Profile().ColorSchemeName(val.Name());
     }
 
-    bool Profiles::BellStyleAudible()
+    bool Profiles::IsBellStyleFlagSet(const uint32_t flag)
+    {
+        // 'flag' is not a compile time constant, so we cannot
+        // use it directly with WI_IsFlagSet and we have to use this
+        // seemingly redundant switch statement instead
+        switch (static_cast<Model::BellStyle>(flag))
+        {
+        case Model::BellStyle::Audible:
+            return WI_IsFlagSet(_State.Profile().BellStyle(), Model::BellStyle::Audible);
+        case Model::BellStyle::Window:
+            return WI_IsFlagSet(_State.Profile().BellStyle(), Model::BellStyle::Window);
+        case Model::BellStyle::Taskbar:
+            return WI_IsFlagSet(_State.Profile().BellStyle(), Model::BellStyle::Taskbar);
+        }
+        return false;
+    }
+
+    void Profiles::SetBellStyleAudible(winrt::Windows::Foundation::IReference<bool> on)
     {
-        return WI_IsFlagSet(_State.Profile().BellStyle(), Model::BellStyle::Audible);
+        auto currentStyle = State().Profile().BellStyle();
+        on.GetBoolean() ? WI_SetFlag(currentStyle, Model::BellStyle::Audible) : WI_ClearFlag(currentStyle, Model::BellStyle::Audible);
```

Comment:
discussed 1:1 -- use WI_UpdateFlag!

Reply:
discussed 1:1 -- make sure you're checking that the IReference _actually contains a bool_.

You can probably use `winrt::unbox_value<bool>` and it will throw if it doesn't contain a bool.